test/integration/test_server.cpp

```diff
@@ -1177,3 +1177,293 @@ TEST_CASE("Test all types with create and describe table") {
     REQUIRE_FALSE(response.table().columns(11).primary_key());
   }
 }
+
+TEST_CASE("AlterTable with constraints", "[integration]") {
+  DestinationSdkImpl service;
+
+  const std::string table_name =
+      "some_table" + std::to_string(Catch::rngSeed());
+  auto token = std::getenv("motherduck_token");
+  REQUIRE(token);
+
+  auto con = get_test_connection(token);
+
+  {
+    // Create Table
+    ::fivetran_sdk::CreateTableRequest request;
+    (*request.mutable_configuration())["motherduck_token"] = token;
+    (*request.mutable_configuration())["motherduck_database"] =
+        TEST_DATABASE_NAME;
+    request.mutable_table()->set_name(table_name);
+    auto col1 = request.mutable_table()->add_columns();
+    col1->set_name("id");
+    col1->set_type(::fivetran_sdk::DataType::STRING);
+    col1->set_primary_key(true);
+
+    auto col2 = request.mutable_table()->add_columns();
+    col2->set_name("name");
+    col2->set_type(::fivetran_sdk::DataType::STRING);
+
+    ::fivetran_sdk::CreateTableResponse response;
+    auto status = service.CreateTable(nullptr, &request, &response);
+    REQUIRE_NO_FAIL(status);
+  }
+
+  {
+    // Alter Table to add a new primary key
+    ::fivetran_sdk::AlterTableRequest request;
+
+    (*request.mutable_configuration())["motherduck_token"] = token;
+    (*request.mutable_configuration())["motherduck_database"] =
+        TEST_DATABASE_NAME;
+    request.mutable_table()->set_name(table_name);
+    auto col1 = request.mutable_table()->add_columns();
+    col1->set_name("id");
+    col1->set_type(::fivetran_sdk::DataType::STRING);
+    col1->set_primary_key(true);
+
+    auto col2 = request.mutable_table()->add_columns();
+    col2->set_name("name");
+    col2->set_type(::fivetran_sdk::DataType::STRING);
+    col2->set_primary_key(true);
```

Comment:
should this be false initially?

Reply:
no, it was "false" in create-table block above. This is turning an existing column that was previously NOT a primary key into a primary key.

Reply:
the test block below `// Alter Table to make an existing (unique valued) column into a primary key` 
has the comment `// turn existing column into a primary key`
```
    auto col2 = request.mutable_table()->add_columns();
    col2->set_name("name");
    col2->set_type(::fivetran_sdk::DataType::STRING);
    col2->set_primary_key(true); // turn existing column into a primary key
```
 so i thought this block should not set this column to a primary key.

Reply:
huh, you are right; I think I duplicated something somewhere

Reply:
I switched it back to false, and refactored to make it a bit easier to read